vm/src/compiler.rs

```diff
@@ -664,7 +665,9 @@ impl<'a> Compiler<'a> {
                 function.emit(ConstructArray(a.expressions.len() as VmIndex));
             }
             Expr::Lambda(ref lambda) => {
-                let (function_index, vars, cf) = try!(self.compile_lambda(&lambda.id,
+                let name = self.symbols.symbol("lambda");
```

Comment:
Maybe keep the `Id` for `Lambda` as it actually serves a purpose there. Having a name for each function is useful in stacktraces.